libbeat/common/kubernetes/watcher.go

```diff
@@ -0,0 +1,263 @@
+package kubernetes
+
+import (
+	"context"
+	"errors"
+	"sync"
+	"time"
+
+	"github.com/elastic/beats/libbeat/common/bus"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"github.com/ericchiang/k8s"
+	corev1 "github.com/ericchiang/k8s/api/v1"
+)
+
+// Watcher reads Kubernetes events and keeps a list of known pods
+type Watcher interface {
+	// Start watching Kubernetes API for new containers
+	Start() error
+
+	// Stop watching Kubernetes API for new containers
+	Stop()
+
+	// Pod returns the running container with the given ID or nil if unknown
+	Pod(ID string) *Pod
```

Comment:
Could you make this `Pod` method more generic? If I want to watch k8s events, it is not a Pod here. I think there should be a more generic interface which should be implemented by resource you want to watch. I have made a PR #5932 and using an `type Resource interface {
	GetMetadata() *ObjectMeta
}
` for this purpose. So in this case, it might be `Resource(id string) Resource`.

Reply:
I like this approach! What do you think about rebasing/reworking this change once this PR is merged?

Reply:
Sure, I will make a PR about this once yours got merged. 😃

Reply:
I found this method may not needed by interface, since pod entity can be observed by bus.Listener?